dotnet/src/SemanticKernel/AI/OpenAI/Clients/OpenAIClientAbstract.cs

```diff
@@ -78,15 +88,16 @@ protected async Task<string> ExecuteCompleteRequestAsync(string url, string requ
     /// <summary>
     /// Asynchronously sends an embedding request for the text.
     /// </summary>
-    /// <param name="url"></param>
-    /// <param name="requestBody"></param>
+    /// <param name="url">URL for the embedding request API</param>
+    /// <param name="requestBody">Text to be embedded</param>
+    /// <param name="httpTimeoutInSeconds">Number of seconds to wait before the request times out</param>
     /// <returns></returns>
     /// <exception cref="AIException"></exception>
-    protected async Task<IList<Embedding<float>>> ExecuteEmbeddingRequestAsync(string url, string requestBody)
+    protected async Task<IList<Embedding<float>>> ExecuteEmbeddingRequestAsync(string url, string requestBody, int httpTimeoutInSeconds = DEFAULT_HTTP_TIMEOUT_IN_SECONDS)
```

Comment:
I'd connect with Abby on how this will be done for embeddings. I'm guessing we'll want to plumb through AIRequestSettings to `GenerateEmbeddingAsync` as well, or use its cancellation token.

Reply:
Hmm I don't think it needs to be that complicated, at least not immediately. Embeddings requests don't have any unique/additional model settings beyond Model and Input strings at the moment. You can set a user, but that has nothing to do with the actual embeddings that get generated or how they get generated.